### PR TITLE
Added the ability to have s-maxage cache control headers (alongside existing max-age cache control headers)

### PR DESCRIPTION
**Pull request by JulieGoldberg**
Added the ability to have s-maxage cache control headers when caching is public, so the length of time something is stored in a CDN can be longer than the length of time it is stored in a browser. e.g. For things stored in AWS CloudFront,  https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html#ExpirationDownloadDist

#### Checklist

- [ X ] run `npm run test` and `npm run benchmark`
- [ X ] tests and/or benchmarks are included
- [ X ] documentation is changed or added
- [ X ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ fastify.listen(3000, (err) => {
 {
   privacy: 'value',
   expiresIn: 300,
+  serverExpiresIn: 2592000,
```
Comment (zekth) [inline Readme.md:100]: This is an advanced feature and i don't think we should describe it here. Also it's incompatible with `privacy: 'value'`

File: plugin.js [hunk 2]
```diff
@@ -69,6 +70,10 @@ function fastifyCachingPlugin (instance, options, next) {
       value = `${_options.privacy}, max-age=${_options.expiresIn}`
     }
 
+    if (_options.privacy.toLowerCase() === 'public' && _options.serverExpiresIn) {
```
Comment (zekth) [inline plugin.js:73]:
```suggestion
    if (_options.privacy !== undefined && _options.privacy.toLowerCase() === 'public' && _options.serverExpiresIn) {
```

File: Readme.md [hunk 3]
```diff
@@ -106,7 +107,8 @@ fastify.listen(3000, (err) => {
 for a *cache-response-directive* as defined by RFC 2616.
 + `expiresIn` (Default: `undefined`): a value, in seconds, for the *max-age* the
 resource may be cached. When this is set, and `privacy` is not set to `no-cache`,
-then `', max-age=<value>'` will be appended to the `cache-control` header.
+then `', max-age=<value>'` will be appended to the `cache-control` header. (300 seconds = 5 minutes)
++ `serverExpiresIn` (Default: `undefined`): a value, in seconds, for the *max-age* the resource may be cached on the server. When this is set, and `privacy` is set to `public`,  then `', s-maxage=<value>'` will be appended to the `cache-control` header. (2,592,000 seconds = 30 days)
```
Comment (zekth) [inline Readme.md:111]: I tend to prefer the explanation from MDN:
```
The s-maxage response directive also indicates how long the response is fresh for (similar to max-age) — but it is specific to shared caches, and they will ignore max-age when it is present.
```
Shared caches makes more sense as it's mostly used on CDN edges.

**Review by zekth — CHANGES_REQUESTED** [review]

File: Readme.md (hunk 3 above)
Comment (JulieGoldberg) [reply Readme.md]: I tried to integrate that without copying their language. Let me know what you think

File: plugin.js (hunk 2 above)
Comment (JulieGoldberg) [reply plugin.js]: Done

File: Readme.md [hunk 4]
```diff
@@ -106,13 +106,14 @@ fastify.listen(3000, (err) => {
 for a *cache-response-directive* as defined by RFC 2616.
 + `expiresIn` (Default: `undefined`): a value, in seconds, for the *max-age* the
 resource may be cached. When this is set, and `privacy` is not set to `no-cache`,
-then `', max-age=<value>'` will be appended to the `cache-control` header.
+then `', max-age=<value>'` will be appended to the `cache-control` header. (300 seconds = 5 minutes)
```
Comment (Fdawgs) [inline Readme.md:109]:
```suggestion
then `', max-age=<value>'` will be appended to the `cache-control` header.
```
Superfluous addition, hopefully readers should know how many seconds are in a minute.

**Review by Fdawgs — COMMENTED** [review]
Nice addition!

File: Readme.md (hunk 4 above)
Comment (JulieGoldberg) [reply Readme.md]: Okay, remove. It was originally in there when I had an example of the time for s-maxage, when I wanted to contrast 5 minutes with a month.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Review by Fdawgs — APPROVED** [review]

**Review by zekth — APPROVED** [review]
lgtm thanks

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by JulieGoldberg** [comment]
Will someone with sufficient permissions merge this? I can't. It's got four approvals. Thanks!

**Pull request merged**